test/jasmine/tests/cone_test.js

```diff
@@ -117,6 +117,55 @@ describe('@gl Test cone autorange', function() {
             _assertAxisRanges('scaled down',
                 [-0.39, 4.39], [-0.39, 4.39], [-0.39, 4.39]
             );
+
+            var trace = fig.data[0];
+
+            var x = trace.x.slice();
+            x.push(5);
+            var y = trace.y.slice();
+            y.push(5);
+            var z = trace.z.slice();
+            z.push(5);
+            var u = trace.u.slice();
+            u.push(0);
+            var v = trace.v.slice();
+            v.push(0);
+            var w = trace.w.slice();
+            w.push(0);
+
+            return Plotly.restyle(gd, {
+                x: [x], y: [y], z: [z],
+                u: [u], v: [v], w: [w]
+            });
+        })
+        .then(function() {
+            _assertAxisRanges('after adding one cone outside range but with norm-0',
+                [-0.45, 6.45], [-0.45, 6.45], [-0.45, 6.45]
```

Comment:
@alexcjohnson unfortunately I can't get this to "stick" to the previous `range[0]` values without breaking other gl3d mocks that depend on

https://github.com/plotly/plotly.js/blob/fb751582506a4d71d23a89786ba8138473e33cd9/src/plots/gl3d/scene.js#L479-L483

which scales the autorange bounds with the `dataScale`:

https://github.com/plotly/plotly.js/blob/fb751582506a4d71d23a89786ba8138473e33cd9/src/plots/gl3d/scene.js#L394-L407

Hopefully this iteration is _close enough_.

--------------

As a side note, we should spend some time replacing this `dataScale` hack. This thing was added to make date axes work properly in 3D. Timestamps don't fit in a Float32Array, so we can't just convert the datestrings to timestamps, but we do this better in `scattergl`, where we use Float64Array or split the data into two Float32Arrays when not supported.

Reply:
New issue about the `dataScale` mess -> https://github.com/plotly/plotly.js/issues/2646